### PR TITLE
DCGAN example

### PR DESCRIPTION
**Pull request by mattya**
This is an example implementation of DCGAN.
Thanks @rezoo for helping me using trainer.

Below is the test result on cifar-10 after 34 epoch.
cifar10

There may be some rooms for improvement to make this example more helpful (e.g. adding celebA dataset example).
Any comments and suggestion will be appreciated.

### CODE REVIEW

**Comment by mattya** [comment]
Thank you for your comments.  I fixed them.

File: examples/dcgan/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# DCGAN
+
+This is an example implementation of DCGAN (https://arxiv.org/abs/1511.06434) using trainer.
+
+This code uses Cifar-10 dataset by default.
+You can use your own dataset by specifying `--dataset` argument to the directory consisting image files for training.
```
Comment (muupan) [inline examples/dcgan/README.md]: I think `consisting of` is correct. You should not remove `of`.

**Comment by muupan** [comment]
LGTM except the tiny grammar fix.

File: examples/dcgan/README.md (hunk 1 above)
Comment (mattya) [inline examples/dcgan/README.md]: Ah, that's right.  Thanks

**Comment by mattya** [comment]
Fixed it

**Pull request merged**